gems/smithy-client/lib/smithy-client/configuration.rb

```diff
@@ -48,6 +48,7 @@ module Client
     #     configuration.build!(price: 100)
     #     #=> raises an ArgumentError, :price was not added as an option
     #
+    # @api private
```

Comment:
I think this is public also.

Reply:
I dont think it needs to be - users never interact with Configuration

Reply:
It is public I think. I believe option() is a shortcut but add_options and after client initialization hooks interact with this?

Reply:
As far as I can tell, only the Configuration::DefaultResolver is actually exposed when developing plugins (through the default block on an option), but I'll remove the api private from it.

After initiailize only gets an instance of the client, which has the generic class for config (ie a resolved config).

Reply:
I believe it lets you conditionally define options based upon if config has an option already. Actually I think this might be quite useful for exposing secondary http config options if http plugin options exist. We probably should have been using this more in v3 development.

Reply:
Ah I see - you mean when defining a Plugin, you can override the `add_options` method which will get [called during client initialization](https://github.com/smithy-lang/smithy-ruby/blob/decaf/gems/smithy-client/lib/smithy-client/base.rb#L64)?

This generally seems like a pretty dangerous way to add options and very order dependent.

Reply:
The intent seems to be that plugins can implement that method and define new config options dynamically which would make this config class public. I'm not sure about ordering. I suppose that makes sense if config is added based on config. It could be based on ruby environment or other components which would not be order dependent.